presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+ * **Description:** Prefer source table layouts that produce streaming operators. Setting this property will allow workers not to wait for chunks of data to start processing them while scanning tables. This may cause faster processing  with lower latency and downtime but some operators may do things more efficiently when working with chunks of data.
+
+
+``push_table_write_through_union``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``optimizer.push-table-write-through-union`` (``true``)
+ * **Description:** See :ref:`optimizer.push-table-writethrough-union <tuning-pref-optimizer>`.
+
+
+``query_max_memory``
+^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``query.max-memory`` (``20 GB``)
+ * **Description:** This property can be use to be nice to the cluster if a particular query is not as important as the usual cluster routines. Setting this value to less than the server property ``query.max-memory`` will cause presto to drop the query in the session if it will require more then ``query_max_memory`` memory. Setting this value to higher than ``query.max-memory`` will not have any effect.
+
+
+
+``query_max_run_time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``query.max-run-time`` (``100 d``)
+ * **Description:** If the expected query processing time is higher than ``query.max-run-time``, it is crucial to set this session property to prevent results of long running queries being dropped after ``query.max-run-time``. A session may also set this value to lower than ``query.max-run-time`` in order to crosscheck for bugs in query. It may be particularly useful when setting up a session with a very large number of short-running queries. It is important to set this value to much higher than the average query time to avoid problems with outliers (some queries may randomly take much longer due to cluster load and other circumstances).
+
+
+``resource_overcommit``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Use resources that are not guaranteed to be available to a query. This property allows you to exceed the limits of memory available per query and session. It may allow resources to be used more efficiently, but may also cause non-deterministic query drops due to insufficient memory on machine. It can be particularly useful for performing more demanding queries.
+
+
+``task_concurrency``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``task.concurrency`` (``1``)
+ * **Description:** Default number of local parallel aggregation jobs per worker. See :ref:`task.concurrency<task-concurrency>`.
+
+
+``task_writer_count``
+^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``task.writer-count`` (``1``)
+ * **Description:** See :ref:`task.writer-count <tuning-pref-task>`.
+
+
 
 JVM Settings
```

Comment:
I believe this is not scope of this PR.
Our goal here was to document more properties not to provide complex tuning guide.

Note that providing tuning guide is a vast task, which has ~unlimited scope (Presto is probably at the point when someone may write a book about that). This has also many layers and probably no single person is qualified to write about all of them.

Even with such a straight forward task as describing properties it was really hard to do this only based on reading code. This is however only a base that may be extended by community in order to provide more informative, concrete and useful guide.

I think this is a big step forward, but obviously it's not a complete guide (I bet it will never be complete guide though).

I also don't think that putting ideas about what may be added here inside a PR is a good idea. I think if you have ideas how to extend this and make it more useful you should either create issue for someone to do or if it is simple tip to add - just do it yourself. The main disadvantage of putting ideas here is not a likely place to look for such informations.

I hope you will understand our point here now, as it was not stated precise enough before. Sorry for that.